tests/test_core/test_generic_section.py

```diff
@@ -128,6 +128,382 @@ def test_rectangular_section():
         )
 
 
+# Test rectangular section tangent stiffness
+@pytest.mark.parametrize(
+    'b, h, E',
+    [
+        (250, 500, 30000),
+        (300, 600, 10000),
+        (400, 200, 20000),
+        (600, 350, 30000),
+    ],
+)
+def test_rectangular_section_tangent_stiffness(b, h, E):
+    """Test stiffness matrix of elastic rectangular section."""
+    # Create materials to use
+    elastic = Elastic(E)
+
+    # The section
+    poly = Polygon(
+        ((-b / 2, -h / 2), (b / 2, -h / 2), (b / 2, h / 2), (-b / 2, h / 2))
+    )
+    geo = SurfaceGeometry(poly, elastic)
+
+    assert geo.polygon.centroid.coords[0][0] == 0
+    assert geo.polygon.centroid.coords[0][1] == 0
+
+    # Create the section with fiber integrator
+    sec = GenericSection(geo, integrator='fiber', mesh_size=0.0001)
+    assert sec.name == 'GenericSection'
+
+    assert math.isclose(sec.gross_properties.area, b * h)
+
+    # compute stiffness matrix
+    stiffness = sec.section_calculator.integrate_strain_profile(
+        [0, 0, 0], 'modulus'
+    )
+    assert stiffness.shape == (3, 3)
+    stiffness /= E
+
+    # Create the section with marin integrator
+    sec = GenericSection(geo)
+    assert sec.name == 'GenericSection'
+
+    assert math.isclose(sec.gross_properties.area, b * h)
+
+    # compute stiffness matrix
+    stiffness = sec.section_calculator.integrate_strain_profile(
+        [0, 0, 0], 'modulus'
+    )
+    assert stiffness.shape == (3, 3)
+    stiffness /= E
```

Comment:
Could this part be even stronger if we parameterize the name of the integrator, i.e. `'fiber', 'marin'`, and check both against the analytical solution?

Reply:
sure!